### ISSUE TITLE
Don't works with RequireJS (optimized)

### ISSUE THREAD

**Issue by KhodeN** [jmdobry/angular-cache#166]
version 4.0.2
override all deps. I think because `module.exports` is used.

Any dependency return 'angular-cache' (string)

**Comment by jmdobry** [comment]
You'll find that `"angular-cache"` is _exactly_ what an angular lib is supposed to return. In fact, as of 1.3.14, [all angular core files themselves export strings](https://github.com/angular/angular.js/blob/master/CHANGELOG.md#features-3). Here's an [article](http://blog.npmjs.org/post/114584444410/using-angulars-new-improved-browserify-support) describing it.

You basically need to be doing something like:

```js
define(['angular', 'angular-cache', function (angular, angularCacheModuleName) {
  angular.module('myApp', [angularCacheModuleName])
    .config(...)
    .run(...);
});
```

**Comment by KhodeN** [comment]
Thx, i do not know about that. 
But this fact no solves the trouble.
`angular-cache` returns for ANY dependency

app.js:
```
define(['angular', 'angular-cache'], function(angular, angularCache){
....
 return 'myApp';
})
```

loader.js
```
require.config({...});
require(['app'], function(app) {
  console.log(app === 'myApp'); // false
  console.log(app === 'angular-cache'); // true WTF!!
});
```

Perhaps WebPack has a bug?

**Comment by KhodeN** [comment]
I will try to reproduce bug in sandbox later

**Comment by jmdobry** [comment]
Now that is quite strange, I'd like to see it reproduced.

**Comment by jmdobry** [comment]
Should be fixed already, but should be definitely fixed in 4.2.1. See the [build_examples](https://github.com/jmdobry/angular-cache/tree/master/build_examples).
